src/annotator/frame-observer.js

```diff
@@ -124,8 +124,9 @@ export function onDocumentReady(frame) {
       frame.hasAttribute('src') &&
       frame.src !== 'about:blank'
     ) {
-      // Unfortunately, listening for 'DOMContentLoaded' on the iframeDocument
-      // doesn't work. Instead, we need to wait for a 'load' event to be triggered.
+      // Listening to `document.DOMContentLoaded` doesn't work because the
+      // document is replaced. On the other hand, listening to `window.load`
```

Comment:
```suggestion
      // document is replaced. On the other hand, listening the frame's `load`
```

It is important to be clear that this refer's to the _frame's_ `load` event, not the load event on a Window (either the `window` global or `frame.contentWindow`). The frame's `contentWindow` changes when the document does. It is only the frame itself that does not change.

Reply:
Thank you for the clarification.

So if I understand this correctly this is the hierarchy:

```
global Window (with load event)
|
|-> document (with a load event)
        | 
        |-> frame (with a load event) ** this is the one that  we need **
                 |
                 |-> window (with a load event)
                        |
                        | - > document (with a load event)
```